chezmoi panics when using with `--exclude scripts`

## Describe the bug

With version v2.0.10, which includes the fix for #1083, the command `chezmoi diff --exclude scripts` panics.

For test purposes I removed the following setting from my config, but it did not change the outcome:

```yaml
diff:
    exclude: ["scripts"]
```

## To reproduce

```
$ chezmoi diff --exclude scripts
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x86c47b]

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*EntryTypeSet).SetSlice(0x0, 0xc0001184f0, 0x1, 0x1, 0x0, 0x0)
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/entrytypeset.go:145 +0x7b
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*EntryTypeSet).Set(0x0, 0x7ffeead05704, 0x7, 0x7, 0xc0002b82f8)
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/entrytypeset.go:117 +0x9e
github.com/spf13/pflag.(*FlagSet).Set(0xc000844600, 0x1033711, 0x7, 0x7ffeead05704, 0x7, 0x7ffeead056fc, 0xc0002b8270)
        /home/runner/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:463 +0xe2
github.com/spf13/pflag.(*FlagSet).Parse.func1(0xc00068fea0, 0x7ffeead05704, 0x7, 0x7, 0xc0002b82f8)
        /home/runner/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1138 +0x54
github.com/spf13/pflag.(*FlagSet).parseLongArg(0xc000844600, 0x7ffeead056fa, 0x9, 0xc000261090, 0x1, 0x1, 0xc00039bb80, 0x20, 0x7fdfacd68108, 0x20, ...)
        /home/runner/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1000 +0x295
github.com/spf13/pflag.(*FlagSet).parseArgs(0xc000844600, 0xc000261090, 0x1, 0x2, 0xc00039bb80, 0x572a78, 0xc000845d00)
        /home/runner/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1108 +0x46e
github.com/spf13/pflag.(*FlagSet).Parse(0xc000844600, 0xc000261080, 0x2, 0x2, 0x0, 0xc00092f9e0)
        /home/runner/go/pkg/mod/github.com/spf13/pflag@v1.0.5/flag.go:1141 +0x13d
github.com/spf13/cobra.(*Command).ParseFlags(0xc0005ae280, 0xc000261080, 0x2, 0x2, 0xc000261080, 0xc00039bcb8)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:1625 +0xbf
github.com/spf13/cobra.(*Command).execute(0xc0005ae280, 0xc000261080, 0x2, 0x2, 0xc0005ae280, 0xc000261080)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:779 +0xae
github.com/spf13/cobra.(*Command).ExecuteC(0xc000690c80, 0xc000690c80, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/twpayne/chezmoi/v2/cmd.(*Config).execute(0xc0002ee800, 0xc00012a010, 0x3, 0x3, 0x0, 0x0)
        /home/runner/work/chezmoi/chezmoi/cmd/config.go:809 +0x7d
github.com/twpayne/chezmoi/v2/cmd.runMain(0x1567c54, 0x6, 0x1587fc0, 0x28, 0x156ae20, 0x14, 0x1568a38, 0xa, 0xc00012a010, 0x3, ...)
        /home/runner/work/chezmoi/chezmoi/cmd/cmd.go:245 +0xe7
github.com/twpayne/chezmoi/v2/cmd.Main(0x1567c54, 0x6, 0x1587fc0, 0x28, 0x156ae20, 0x14, 0x1568a38, 0xa, 0xc00012a010, 0x3, ...)
        /home/runner/work/chezmoi/chezmoi/cmd/cmd.go:85 +0x8e
main.main()
        /home/runner/work/chezmoi/chezmoi/main.go:17 +0x131
```

## Expected behavior

`--exclude scripts` should work the same as before v2.0.10.

## Output of command with `--verbose`

Same output as without `--versbose`

## Output of `chezmoi doctor`

Not relevant, everything is ok.

--- a/cmd/config.go
+++ b/cmd/config.go
@@ -204,6 +204,7 @@ func newConfig(options ...configOption) (*Config, error) {
 		Umask:   chezmoi.Umask,
 		Color:   "auto",
 		Diff: diffCmdConfig{
+			Exclude: chezmoi.NewEntryTypeSet(chezmoi.EntryTypesNone),
 			Pager:   os.Getenv("PAGER"),
 			include: chezmoi.NewEntryTypeSet(chezmoi.EntryTypesAll),
 		},

--- a/cmd/diffcmd_test.go
+++ b/cmd/diffcmd_test.go
@@ -1,0 +1,104 @@
+package cmd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twpayne/go-vfs/v2"
+	"github.com/twpayne/go-vfs/v2/vfst"
+
+	"github.com/twpayne/chezmoi/v2/internal/chezmoitest"
+)
+
+func TestDiffCmd(t *testing.T) {
+	if chezmoitest.Umask != 0o22 {
+		t.Skip("umask not 0o22")
+	}
+	for _, tc := range []struct {
+		name      string
+		extraRoot interface{}
+		args      []string
+		stdoutStr string
+	}{
+		{
+			name: "empty",
+		},
+		{
+			name: "file",
+			extraRoot: map[string]interface{}{
+				"/home/user/.local/share/chezmoi": map[string]interface{}{
+					"dot_file": "# contents of .file\n",
+				},
+			},
+			stdoutStr: chezmoitest.JoinLines(
+				`diff --git a/.file b/.file`,
+				`new file mode 100644`,
+				`index 0000000000000000000000000000000000000000..8a52cb9ce9551221716a53786ad74104c5902362`,
+				`--- /dev/null`,
+				`+++ b/.file`,
+				`@@ -0,0 +1 @@`,
+				`+# contents of .file`,
+			),
+		},
+		{
+			name: "simple_exclude_files",
+			extraRoot: map[string]interface{}{
+				"/home/user/.local/share/chezmoi": map[string]interface{}{
+					"dot_file":            "# contents of .file\n",
+					"symlink_dot_symlink": ".file\n",
+				},
+			},
+			args: []string{
+				"--exclude", "files",
+			},
+			stdoutStr: chezmoitest.JoinLines(
+				`diff --git a/.symlink b/.symlink`,
+				`new file mode 120000`,
+				`index 0000000000000000000000000000000000000000..3e6844d17780d623d817c3e22bcd1128d64422ae`,
+				`--- /dev/null`,
+				`+++ b/.symlink`,
+				`@@ -0,0 +1 @@`,
+				`+.file`,
+			),
+		},
+		{
+			name: "simple_exclude_files_with_config",
+			extraRoot: map[string]interface{}{
+				"/home/user": map[string]interface{}{
+					".config/chezmoi/chezmoi.toml": chezmoitest.JoinLines(
+						`[diff]`,
+						`  exclude = ["files"]`,
+					),
+					".local/share/chezmoi": map[string]interface{}{
+						"dot_file":            "# contents of .file\n",
+						"symlink_dot_symlink": ".file\n",
+					},
+				},
+			},
+			stdoutStr: chezmoitest.JoinLines(
+				`diff --git a/.symlink b/.symlink`,
+				`new file mode 120000`,
+				`index 0000000000000000000000000000000000000000..3e6844d17780d623d817c3e22bcd1128d64422ae`,
+				`--- /dev/null`,
+				`+++ b/.symlink`,
+				`@@ -0,0 +1 @@`,
+				`+.file`,
+			),
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			chezmoitest.WithTestFS(t, map[string]interface{}{
+				"/home/user": &vfst.Dir{Perm: 0o777 &^ chezmoitest.Umask},
+			}, func(fs vfs.FS) {
+				if tc.extraRoot != nil {
+					require.NoError(t, vfst.NewBuilder().Build(fs, tc.extraRoot))
+				}
+				var stdout strings.Builder
+				require.NoError(t, newTestConfig(t, fs, withStdout(&stdout)).execute(append([]string{"diff"}, tc.args...)))
+				assert.Equal(t, tc.stdoutStr, stdout.String())
+			})
+		})
+	}
+}

--- a/internal/chezmoi/entrytypeset.go
+++ b/internal/chezmoi/entrytypeset.go
@@ -192,7 +192,7 @@ func (s *EntryTypeSet) Type() string {
 // from a []string.
 func StringSliceToEntryTypeSetHookFunc() mapstructure.DecodeHookFunc {
 	return func(from, to reflect.Type, data interface{}) (interface{}, error) {
-		if to != reflect.TypeOf(&EntryTypeSet{}) {
+		if to != reflect.TypeOf(EntryTypeSet{}) {
 			return data, nil
 		}
 		sl, ok := data.([]interface{})

--- a/testdata/scripts/exclude.txt
+++ b/testdata/scripts/exclude.txt
@@ -1,5 +1,10 @@
 [!umask:022] skip
 
+# test chezmoi diff --exclude
+chezmoi diff --exclude=scripts
+[!windows] cmp stdout golden/diff-no-scripts
+[windows] cmp stdout golden/diff-no-scripts-windows
+
 # test that chezmoi diff respects the diff.exclude configuration variable
 chezmoi diff
 [!windows] cmp stdout golden/diff

